lib/array_equals.rb

```diff
@@ -1,5 +1,16 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
-def array_equals(array1, array2)
-  raise NotImplementedError
+def array_equals(arr1, arr2)
+  if (arr1 == [] && arr2 == [] || arr1 == nil && arr2 == nil)
+    return true
+  elsif arr1 == nil || arr2 == nil || arr1.length != arr2.length
+    return false 
```

Comment:
Good job on guarding for edge cases and exiting early if a required condition is met.